matrix-notifications.el

```diff
@@ -0,0 +1,58 @@
+
+(defvar matrix-client-notify-hook
+  '(matrix-client-notify)
+  "List of functions called for events.
+Each is called with the event-type and the event data.")
+
+(defvar matrix-client-notifications nil
+  "Alist of recent notifications mapping notification ID to related buffer.
+Automatically trimmed to last 20 notifications.")
+
+(defun matrix-client-notify (event-type data &rest rest)
+  "Notify for an event of EVENT-TYPE with DATA."
+  (let ((fn (intern-soft (concat "matrix-client-notify-" event-type))))
+    (when (functionp fn)
+      (apply #'funcall fn data rest))))
+
+;; MAYBE: Use a macro to define the handlers, because they have to
+;; define their arg lists in a certain way, and the macro would take
+;; care of that automatically.
+
+(cl-defun matrix-client-notify-m.room.message (data &key room &allow-other-keys)
+  "Show notification for m.room.message events.
+DATA should be the `data' variable from the
+`defmatrix-client-handler'.  ROOM should be the room object."
+  (pcase-let* (((map content sender event_id) data)
+               ((map body) content)
+               (display-name (matrix-client-displayname-from-user-id room sender))
+               (buffer (oref room :buffer))
+               (id (notifications-notify :title (format "<b>%s</b>" display-name)
```

Comment:
I don't like this behavior too much actually, since we would need to add a bunch of variables (potentially) for configuring the notification behavior. [I've used alert.el](https://github.com/jwiegley/alert) in the past, and it solves all of those problems, by having us send notifications to alert, and the user can configure them in alert by themselves. I don't think it would be too hard to switch to using alert insetad of `notifications-notify`, what do you think about it? I think the other emacs chat clients use alert as well.

For example, if someone wanted notifications in the message buffer, that could be done easily if we used alert, but not so much if we use `notifications-notify`.

Reply:
Sounds like a good idea.  I didn't know about alert, but I'll check it out.  I guess Sauron is another option, but from looking at the code in matrix-sauron, it looks pretty complicated.